acadopy/bindings.pyx

```diff
@@ -404,24 +436,68 @@ cdef class DifferentialState(ExpressionType):
     """ Python wrapper of the DifferentialState class
     """
 
-    def __cinit__(self, str name=None, int nrows=0, int ncols=0, initialize=True):
+    def __cinit__(self, _arg=None, int nrows=1, int ncols=1, initialize=True):
+        """ Support a combination of constructors 
+        
+        - ()
+        - (name, nrows, ncols) 
+        
+        """
+    
         if type(self) is DifferentialState:
             if initialize:
-                if name is None:
-                    self._thisptr = new acado.DifferentialState()
-                else:
+                if _arg is not None:
                     self._thisptr = new acado.DifferentialState(
-                        name, <unsigned int>nrows, <unsigned int>ncols
+                        <str>_arg, <unsigned int>nrows, <unsigned int>ncols
                     )
+                else:
+                    self._thisptr = new acado.DifferentialState()       
+                self._owner = True
+            else:
+                self._owner = False
 
 cdef class IntermediateState(ExpressionType):
     """ Python wrapper of the IntermediateState class
     """
 
-    def __cinit__(self, initialize=True):
+    def __cinit__(self, _arg=None, int nrows=1, int ncols=1, initialize=True):
```

Comment:
Referencing elements of an `IntermediateState` other than for assignment raises:

`TypeError: 'acadopy.bindings.IntermediateState' object is not subscriptable`

This could be an issue if we want to use it as a container for generic acado-type objects to pass into a solver, since we may want to implement constraints:

  `a = IntermediateState(1)`
  `a[0] = Parameter()`
 `a_solver.subjectTo(lower_value <= a[0])`

But currently the `<=` operator is not supported for `IntermediateState` class.

Reply:
__getitem__ is now implemented